securedrop/tests/test_journalist.py

```diff
@@ -257,111 +260,92 @@ def test_admin_edits_user_password_success_response(self):
         self._login_admin()
         resp = self.client.post(url_for('admin_add_user'),
                                 data=dict(username=self.admin.username,
-                                          password='testtesttest',
-                                          password_again='testtesttest',
+                                          password=VALID_PASSWORD,
                                           is_admin=False))
         self.assertIn('That username is already in use', resp.data)
 
     def test_max_password_length(self):
         """Creating a Journalist with a password that is greater than the
         maximum password length should raise an exception"""
-        overly_long_password = 'a'*(Journalist.MAX_PASSWORD_LEN + 1)
+        overly_long_password = VALID_PASSWORD + \
+            'a' * (Journalist.MAX_PASSWORD_LEN - len(VALID_PASSWORD) + 1)
         with self.assertRaises(InvalidPasswordLength):
             Journalist(username="My Password is Too Big!",
                        password=overly_long_password)
 
     def test_min_password_length(self):
         """Creating a Journalist with a password that is smaller than the
-        minimum password length should raise an exception"""
+           minimum password length should raise an exception. This uses the
+           magic number 7 below to get around the "diceware-like" requirement
+           that may cause a failure before the length check.
+        """
+        password = ('a ' * 7)[0:(Journalist.MIN_PASSWORD_LEN - 1)]
         with self.assertRaises(InvalidPasswordLength):
             Journalist(username="My Password is Too Small!",
-                       password='tiny')
+                       password=password)
 
     def test_admin_edits_user_password_too_long_warning(self):
         self._login_admin()
-        overly_long_password = 'a' * (Journalist.MAX_PASSWORD_LEN + 1)
+        overly_long_password = VALID_PASSWORD + \
+            'a' * (Journalist.MAX_PASSWORD_LEN - len(VALID_PASSWORD) + 1)
 
-        self.client.post(
-            url_for('admin_edit_user', user_id=self.user.id),
+        resp = self.client.post(
+            url_for('admin_new_password', user_id=self.user.id),
             data=dict(username=self.user.username, is_admin=False,
-                      password=overly_long_password,
-                      password_again=overly_long_password),
+                      password=overly_long_password),
             follow_redirects=True)
 
-        self.assertMessageFlashed('Your password must be between {} and {} '
-                                  'characters.'.format(
-                                      Journalist.MIN_PASSWORD_LEN,
-                                      Journalist.MAX_PASSWORD_LEN), 'error')
+        print resp.data.decode('utf-8')
+        self.assertMessageFlashed('You submitted a bad password! '
+                                  'Password not changed.', 'error')
```

Comment:
Because there is nothing that should be able to happen on accident and all of the password submission/etc is outside of the user's control. More detailed messages won't provide actionable steps.